lapixdl/evaluation/model.py

```diff
@@ -614,22 +618,22 @@ def iou(self) -> float:
         return self._iou
 
     @property
-    def precision_recall_curve(self) -> List[Tuple[float, float]]:
+    def precision_recall_curve(self) -> list[tuple[float, float]]:
         """List[Tuple[float, float]]: Precision x Recall curve as a list of (Recall, Precision) tuples."""
-        assert self.gt_count > 0, "This class does not have instances."
+        assert self.gt_count > 0, 'This class does not have instances.'
         return self._precision_recall_curve
 
-    def average_precision(self, interpolation_points: Optional[int] = None) -> float:
+    def average_precision(self, interpolation_points: int | None = None) -> float:
         """Calculates the Average Precision metric.
 
         Args:
-            interpolation_points (Optional[int], optional): Number of points to use for interpolation. 
+            interpolation_points (Optional[int], optional): Number of points to use for interpolation.
             Uses all points if None. Defaults to None.
 
         Returns:
             float: Average Precision metric.
         """
-        if not interpolation_points is None:
+        if interpolation_points is not None:
             return self._interpolated_average_precision(interpolation_points)
         else:
             return self._average_precision()
```

Comment:
I think this is what codefactor want you to do:
```suggestion
            return self._interpolated_average_precision(interpolation_points)
        return self._average_precision()
```

If you think it is better to keep the "else" statement, I think we should disable the `no-else-return` rule.